query-planner-js/src/buildPlan.ts

```diff
@@ -3628,6 +3628,14 @@ function maybeSubstratPathPrefix(basePath: OperationPath, maybePrefix: Operation
   return undefined;
 }
 
+function updateCreatedGroups(createdGroups: FetchGroup[], ...newCreatedGroups: FetchGroup[]) {
```

Comment:
So if I understand this correctly, the issue was that if a FetchGroup was present multiple times, the hander would not process the duplicates, leading to an assertion?